pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -683,12 +698,14 @@ public void setupInput(ContextImpl contextImpl) throws Exception {
         }
         this.source = (Source<?>) object;
 
+        Thread.currentThread().setContextClassLoader(this.functionClassLoader);
         if (sourceSpec.getConfigs().isEmpty()) {
             this.source.open(new HashMap<>(), contextImpl);
         } else {
             this.source.open(new Gson().fromJson(sourceSpec.getConfigs(),
                     new TypeToken<Map<String, Object>>(){}.getType()), contextImpl);
         }
+        Thread.currentThread().setContextClassLoader(this.frameworkClassLoader);
```

Comment:
Same here, we should use `try / finally` to make sure it goes back to frameworkClassLoader

Reply:
will fix